lib/elixir_sense/core/state.ex

```diff
@@ -188,24 +190,36 @@ defmodule ElixirSense.Core.State do
   def escape_protocol_impementations({protocol, implementations}) do
     joined_implementations =
       implementations
-      |> Enum.map(fn parts ->
-        parts
-        |> Enum.map(&Atom.to_string/1)
-        |> Enum.join(@dot_marker)
+      |> Enum.map(fn
+        parts when is_list(parts) ->
+          parts
+          |> Enum.map(&Atom.to_string/1)
```

Comment:
Can use `Enum.map_join(parts, @dot_marker, &Atom.to_string/`)` to save an iteration.